tests/_regtest_outputs/test_run_topostats.test_process_scan_both.out

```diff
@@ -1,24 +1,23 @@
-17               460.815289  458.615292    8.335568   28.018485    16.951505      16.480915    0.746482    3.152826       2.161743     1.991000  1282.726954  644.262682          1616.513637                21.778540                 52.651772             1146.678695      2.417599     upper  53.354603  21.778540        69.814554      True             0.000000  minicircle.spm
-18               419.711981  456.335815   10.647572   24.867679    18.230207      18.820515    0.746054    3.046720       2.133687     1.976641  1370.916403  693.558538          1625.299038                35.074208                 42.978255             1507.428252      1.225352     upper  45.966170  33.746032       101.819813      True             0.000000  minicircle.spm
-19                90.924690  469.014736    4.419163   25.241880    16.115111      16.417253    0.746537    2.936420       2.109230     1.958485  1300.014367  663.785793          1600.163031                21.150850                 49.017251             1036.756498      2.317507     upper  49.083113  21.150850       104.650350     False            14.128829  minicircle.spm
-20                19.828838  501.350732    2.045679   25.469797    16.062272      16.647287    0.745369    2.989892       2.126276     1.971355  1210.415330  614.001859          1366.617809                20.435630                 50.047841             1022.759152      2.449048     upper  50.157721  20.435630       102.537579     False            20.559070  minicircle.spm
+                threshold    centre_x    centre_y  radius_min  radius_max  radius_mean  radius_median  height_min  height_max  height_median  height_mean       volume        area  area_cartesian_bbox  smallest_bounding_width  smallest_bounding_length  smallest_bounding_area  aspect_ratio  max_feret  min_feret      Image Name  Contour Lengths Circular  End to End Distance
+Molecule Number                                                                                                                                                                                                                                                                                                                                                                      
+0                   upper   18.739027   96.544583    5.914298   20.891859    15.564967      16.245908    0.776713    3.922363       2.093704     1.999236  1274.860013  637.673631          1137.465287                24.428348                 40.849626              997.888853      1.672222  41.334203  24.232548  minicircle.spm        46.508479     True             0.000000
```

Comment:
I have changed the `threshold_std_dev_lower` value in the `grains` section of `default_config.yaml` to be `1.0`. This should cause a grain to be detected, and therefore fix the issue you raised there. A potential problem comes with the fact that I cannot change the value in the `filters` section of the code, because if I do, then this will make the flattening worse. The value for the flattening section is `10.0` as to ensure that no data will be detected for lower parts by default, since by default we want it to be configured for AFM images where we only want to see and are expecting to see high up grains. 

Perhaps I should make a `direction` parameter like with `grains` to enable / disable upper / lower threshold values instead of setting the default lower one to filter out 99.9999..% of the data.

Reply:
Completely understand going with the default and what is most likely to be used.

One of the aspects of testing though is to try and capture edge cases. The lower threshold was introduced for a reason, my memory is hazy but perhaps something to do with scans of silicone, and so we should have tests that cover that too.

Modifying parameters in tests is what we should do to cover this and its fairly straight-forward to setup fixtures of configs with modified values. The `process_scan_config()` fixture defined on lines 44-53 of `tests/conftest.py` shows how to do this.

However, if `Filters()` will _never_ use a lower threshold then there is no need to even have a `threshold_std_dev_lower` in there to be tested.

Reply:
@ns-rse I updated the config used for the `process_scan` tests like you said in the `process_scan_config` fixture and the tests work well now while maintaining the correct values in `default_config.yaml` 

The lower thresholding capabilities are needed as they can be useful for other sample types.